### PR TITLE
Feature add monitoring to firewall policy

### PR DESCRIPTION
**Pull request by jk-ans**
# Description

Please provide a summary of your changes and the reason the changes are being made.

Adding Policy Insights to Firewall Policy.

Also adding some variables to the Firewall module.

Please select the option that is relevant.

  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [x] This change requires a documentation update

# Checklist:

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my code
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings e.g. code analysis tooling or general use of the changed code

### CODE REVIEW

File: azure/firewall/variables.tf
Comment (djackson-ans) [inline azure/firewall/variables.tf:78]: Perhaps this can be validated somehow?

Comment (jk-ans) [reply azure/firewall/variables.tf:78]: added CIDR validation

File: azure/firewall-policy/main.tf
Comment (jk-ans) [reply azure/firewall-policy/main.tf:22]: Changed to var.threat_intelligence_allowlist.ip_addresses & var.threat_intelligence_allowlist.fqdns

File: azure/firewall/variables.tf
Comment (djackson-ans) [reply azure/firewall/variables.tf:78]: Ahh, sorry, I wasn't clear.

I meant validate the input so that it only accepts a list of RFC1918 addresses?

EG:
https://stackoverflow.com/questions/73860207/terraform-regex-how-to-validate-ipv4-or-private-cidr-as-per-the-rfc1918

It's more of a nice to have. I'll approve the change anyway, but if you've got time, it makes sense to add this in before merging.

**Review by djackson-ans — APPROVED** [review]
Approved with comments

File: azure/firewall/variables.tf
Comment (jk-ans) [reply azure/firewall/variables.tf:78]: created an issue to look at this later

**Pull request merged**
